src/couch_changes.erl

```diff
@@ -377,6 +379,21 @@ check_selector(_Selector) ->
     throw({bad_request, "Selector error: expected a JSON object"}).
 
 
+check_fields(nil) ->
+    nil;
+check_fields(Fields) when is_list(Fields) ->
+    try
+        {ok, Fields1} = mango_fields:new(Fields),
+        Fields1
+    catch
+        {mango_error, Mod, Reason0} ->
+            {_StatusCode, _Error, Reason} = mango_error:info(Mod, Reason0),
+            throw({bad_request, Reason})
```

Comment:
It does, it throws `{mango_error,...` but in this context other validation functions return `{bad_request,...` so made mango fields parsing do the same